bashbot.sh

```diff
@@ -321,15 +328,37 @@ forward() {
 	res=$(curl -s "$FORWARD_URL" -F "chat_id=$1" -F "from_chat_id=$2" -F "message_id=$3")
 }
 
+
+background() {
+	echo "${CHAT[ID]}:$2:$1" >"$TMPDIR/${copname}$2-back.cmd"
+	startproc "$1" "back-$2-"
+}
+
 startproc() {
-	killproc
-	mkfifo /tmp/$copname
-	TMUX= tmux new-session -d -s $copname "$* &>/tmp/$copname; echo imprettydarnsuredatdisisdaendofdacmd>/tmp/$copname"
-	TMUX= tmux new-session -d -s sendprocess_$copname "bash $SCRIPT outproc ${CHAT[ID]} $copname"
+	killproc "$2"
+	local fifo="$2${copname}" # add $1 to copname, so we can have more than one running script per chat
+	mkfifo "$TMPDIR/${fifo}"
+	TMUX= tmux new-session -d -s "${fifo}" "$1 &>$TMPDIR/${fifo}; echo imprettydarnsuredatdisisdaendofdacmd>$TMPDIR/${fifo}"
```

Comment:
In this line I use $1 instead of $* to run the script, this may lead to incompatibilities with startproc callers using multiple arguments when callung 'startproc'.